cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -100,3 +102,14 @@ func makeModelKeyValue(agentTags map[string]string) []model.KeyValue {
 
 	return tags
 }
+
+func (r *Reporter) Retryable(err error) bool {
```

Comment:
Instead of introducing this introspection method to the Reporter interface, I think it would be cleaner to encode this directly into the error returned.
```go
// in the top reporter.go file
type RetryableError interface {
    error
    IsRetryable() bool
}

func IsRetryable(err error) bool {
    if r, ok := err.(RetryableError); ok {
        return r.IsRetryable()
    }
    return false
}

// in grpc reporter
type maybeRetryableGRPCError error

func (r maybeRetryableGRPCError) IsRetryable() bool {
   // this logic
}
```

Then gRPC reporter just returns `maybeRetryableGRPCError(err)`